### ISSUE TITLE
Double urldecoding in ConfirmRegister::verifyManifest breaks activation for certain users.

### ISSUE THREAD

**Issue by simplesmiler** [modxcms/Login#38]
From [/core/components/login/controllers/web/ConfirmRegister.php](https://github.com/modxcms/Login/blob/8913f66aafd95ba6819dd0f4e07c64fa3b1a024b/core/components/login/controllers/web/ConfirmRegister.php#L75-L90):
```
$this->username = base64_decode(urldecode(rawurldecode($_REQUEST['lu'])));
$this->password = base64_decode(urldecode(rawurldecode($_REQUEST['lp'])));
```

If `base64_encode(username)` contains `+`, then `urldecode` will turn this plus into a space, which in turn breaks decoding username wth `base64_decode`.

**Comment by gadgetto** [comment]
Wow this is a good catch! I already ripped my hair off because of this activation bug!
Any idea for a workaround?

**Comment by gadgetto** [comment]
Seems that there are pretty good methods for URL safe encoding included in this package which aren't used:

https://github.com/modxcms/Login/blob/develop/core/components/login/model/login/login.class.php#L272

**Comment by gadgetto** [comment]
Also found this in PHP Docs:
http://php.net/manual/de/function.base64-encode.php#103849

```
function base64url_encode($data) { 
    return rtrim(strtr(base64_encode($data), '+/', '-_'), '='); 
} 

function base64url_decode($data) { 
    return base64_decode(str_pad(strtr($data, '-_', '+/'), strlen($data) % 4, '=', STR_PAD_RIGHT)); 
} 
```

**Comment by simplesmiler** [comment]
From my experience around 1/4 of users registered were not able to activate their accounts.

At the time I worked around this issue with the following:
``` php
$this->username = base64_decode(rawurldecode($_REQUEST['lu']));
$this->password = base64_decode(rawurldecode($_REQUEST['lp']));
```

`base64url_encode` and `base64url_decode` are better, but will probably require carefully splattering them throughout the Login code.

**Comment by gadgetto** [comment]
I fixed the problem in this pull request:
https://github.com/modxcms/Login/pull/47

(fully tested and works!)

**Comment by gadgetto** [comment]
Fixed in 966044d. Can be closed.

**Comment by simplesmiler** [comment]
:+1:
